node/public.go

```diff
@@ -0,0 +1,66 @@
+// Package node provides a high level wrapper around tendermint services.
+package node
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/service"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/privval"
+	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/types"
+)
+
+// NewDefault constructs a tendermint node service for use in go process that
+// host their own process-local tendermint node. This is roughly
+// equivalent to running tendermint in it's own process communicating
+// to an external ABCI application.
+func NewDefault(conf *config.Config, logger log.Logger) (service.Service, error) {
+	return newDefaultNode(conf, logger)
+}
+
+// New constructs a tendermint node for use in an integrated
+// context. The ClientCreator makes it possible to construct.
+// The final option is a pointer to a Genesis document: if the value
+// is nil, the genesis document is read from the file specified in the
+// config, and otherwise the node uses value of the final argument.
+func New(conf *config.Config,
```

Comment:
So my view on this is that we somewhat have two configs. One is the native go config which includes the application, the genesis doc, db providers and what not and then we have the toml config which basically parses strings to and from the config.toml. Then we have a function that takes the toml config and creates the native go config. The node constructor takes only the native go config - thus we accomodate to both applications that want to run in and out of process. Does that make sense?

Reply:
I'm not sure I understand what the change you're suggesting we make is? 

I think having different constructor functions for different use cases probably makes things easier from the user perspective, but I'm definitely down to change this.

Reply:
resolving after a conversation with @cmwaters , who promises to open a PR to follow this one with this change. (which I think is correct, but I can't quite see it and I don't think it makes sense to hold up this work on that.